rflx/pyrflx/typevalue.py

```diff
@@ -463,161 +535,153 @@ def accepted_type(self) -> type:
     def literals(self) -> Mapping[Name, Expr]:
         return {}
 
-    def assign(self, value: bytes, length: int = 0, check: bool = True) -> None:
+    @property
+    def size(self) -> Number:
+        return Number(len(self.bitstring))
+
+    def assign(self, value: bytes, check: bool = True) -> None:
 
         msg_as_bitstr: Bitstring = Bitstring().from_bytes(value)
-        self.assign_bitvalue(msg_as_bitstr, length)
+        self.parse(msg_as_bitstr)
```

Comment:
I think an `assign` method for `MessageValue` is very confusing. There is no equivalent message object in python like for the other `TypeValue`. I would prefer replacing the implementation by an exception.

Reply:
Do you mean removing the functionality to parse a message from a bytestring altogether or implementing this in a differently named method?

Reply:
Maybe `parse` could be generally changed to allow `Bitstring` as well as `bytes` as `value`. It should be quite simple to check if `value` is of type `bytes` and convert it in that case.

Reply:
Fixed by aa19463